src/main/java/io/rebloom/client/Client.java

```diff
@@ -358,4 +366,271 @@ private Connection sendCommand(Jedis conn, ProtocolCommand command, byte[]... ar
+    try (Jedis conn = _conn()) {
+      return sendCommand(conn, CuckooCommand.COUNT, //
+          SafeEncoder.encode(key), //
+          SafeEncoder.encode(item) //
+      ).getIntegerReply();
+    }
+  }
+
+  @Override
+  public IteratorDataPair cfScanDump(String key, long iteratorValue) {
+    try (Jedis conn = _conn()) {
+      List<Object> response = sendCommand(conn, CuckooCommand.SCANDUMP, //
+          SafeEncoder.encode(key), //
+          Protocol.toByteArray(iteratorValue) //
+      ).getObjectMultiBulkReply();
+
+      return new IteratorDataPair(((Long) response.get(0)).longValue(), (byte[]) response.get(1));
+    }
+  }
+
+  @Override
+  public void cfLoadChunk(String key, IteratorDataPair idp) {
+    try (Jedis conn = _conn()) {
+      String rep = sendCommand(conn, CuckooCommand.LOADCHUNK, //
+          SafeEncoder.encode(key), //
+          Protocol.toByteArray(idp.getIteratorValue()), //
+          idp.getData() //
+      ).getStatusCodeReply();
+
+      if (!rep.equals("OK")) {
+        throw new JedisException(rep);
+      }
+    }
+  }
+
+  @Override
+  public Map<String, Long> cfInfo(String key) {
+    try (Jedis conn = _conn()) {
+      List<Object> values = sendCommand(conn, CuckooCommand.INFO, SafeEncoder.encode(key)).getObjectMultiBulkReply();
+
+      Map<String, Long> infoMap = new HashMap<String, Long>(values.size() / 2);
+      for (int i = 0; i < values.size(); i += 2) {
+        Long val = (Long) values.get(i + 1);
+        infoMap.put(SafeEncoder.encode((byte[]) values.get(i)), val);
+      }
+      return infoMap;
+    }
+  }
+
+  @Override
+  public Iterator<IteratorDataPair> cfScanDumpIterator(String key) {
+    return new CfScanDumpIterator(this, key);
+  }
+
+  @Override
+  public Stream<IteratorDataPair> cfScanDumpStream(String key) {
+    return StreamSupport.stream( //
+        Spliterators.spliteratorUnknownSize( //
+            new CfScanDumpIterator(this, key), //
+            Spliterator.ORDERED //
+        ), false);
+  }
+
+  /**
+   * An Iterator over the elements (IteratorDataPair) obtained for a Cuckoo Filter
+   * using CF.SCANDUMP.
+   *
+   */
+  private static class CfScanDumpIterator implements Iterator<IteratorDataPair> {
+
+    private Client client;
+    private String key;
+    private IteratorDataPair current;
+    private boolean firstAccess = true;
+
+    /**
+     * Create a CfScanDumpIterator for a given Cuckoo Filter using an instance of
+     * the Client
+     * 
+     * @param client RedisBloom Filter client
+     * @param key    Name of the filter
+     */
+    public CfScanDumpIterator(Client client, String key) {
+      this.client = client;
+      this.key = key;
+    }
+
+    @Override
+    public boolean hasNext() {
+      if (firstAccess) {
+        current = pullNext();
+      }
+
+      return current != null;
+    }
+
+    @Override
+    public IteratorDataPair next() {
+      IteratorDataPair result = current;
+      current = pullNext();
```

Comment:
Instead of pulling the next element eagerly, can we pull it when it is *actually* needed?

Reply:
@sazzad16 I tried to find a clean way to do that, but I ran into needing to check for zero return to be able to answer the `hasNext()` question and it felt dirty to pull the next result on `hasNext`, feels like a getter with side-effects but I guess we can duplicate the call just to answer the question. Any ideas you have please share!

Reply:
@bsbodden My two cents is that it is better to pull an element when checking its existence instead of pulling it during previous element's usage.

Reply:
@sazzad16 Thanks, I just needed somebody to bounce of ideas, I'll adjust to make it eager and just fire a call to on `hasNext()`.